evmcore/state_processor_test.go

```diff
@@ -359,11 +362,14 @@ func TestApplyTransaction_ApplyMessageError_RevertsSnapshotIfPrague(t *testing.T
 
 			state.EXPECT().GetBalance(msg.From).Return(uint256.NewInt(1000000))
 			state.EXPECT().SubBalance(any, any, any)
+			state.EXPECT().EndTransaction()
 
 			if isPrague {
 				// Set up snapshot and revert expectations
-				state.EXPECT().Snapshot().Return(42)
-				state.EXPECT().RevertToSnapshot(42)
+				gomock.InOrder(
+					state.EXPECT().Snapshot().Return(42),
+					state.EXPECT().RevertToSnapshot(42),
```

Comment:
Please add to the expectation that the mutating operation `SubBalance` is indeed between the creation of the snapshot and its revert.